examples/repl/main.rs

```diff
@@ -387,15 +387,15 @@ impl Highlighter for DcHelper {
 impl Helper for DcHelper {}
 
 fn main_0(args: Vec<String>) -> Result<(), failure::Error> {
-    let mut context = dc_context_new(Some(receive_event), ptr::null_mut(), Some("CLI".into()));
-
-    if args.len() == 2 {
-        if unsafe { !dc_open(&mut context, &args[1], None) } {
-            println!("Error: Cannot open {}.", args[0],);
-        }
-    } else if args.len() != 1 {
+    if args.len() < 2 {
         println!("Error: Bad arguments, expected [db-name].");
+        return Err(format_err!("No db-name specified"));
     }
+    let context = Context::new(
+        Box::new(receive_event),
+        "CLI".into(),
+        Path::new(&args[1]).to_path_buf(),
```

Comment:
is the `to_path_buf` really needed, couldn't the method take `impl AsRef<Path>` instead?

Reply:
I considered this too but thought that making the Context own all off it's data was better/easier so decided to go with that.  I think it may make some things easier wrt moving things across threads or whatever (not that i have any specific plans currently)